### PR TITLE
Calculated attributes one chain

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: server/modules/analysis/repository/processingStepCalculationRepository.js [hunk 1]
```diff
@@ -89,6 +92,27 @@ export const fetchCalculationAttributeUuidsByChainUuid = async (surveyId, chainU
     R.prop('result'),
   )
 
+/**
+ * Returns an array of calculation attribute definition uuids used by chains different from the specified one
+ */
+export const fetchCalculationAttributeUuidsOtherChainsByChainUuid = async (surveyId, chainUuid, client = db) =>
```
Comment (minotogna) [inline server/modules/analysis/repository/processingStepCalculationRepository.js]: can this method be ```fetchCalculationAttributeUuids = async (surveyId, chainUuidExcluded, client = db) ```

File: webapp/loggedin/modules/analysis/processingChain/actions.js [hunk 2]
```diff
@@ -37,22 +37,38 @@ export const navigateToProcessingChainsView = history => dispatch => {
   history.push(appModuleUri(analysisModules.processingChains))
 }
 
+export const onProcessingChainCreateOrFetch = processingChain => async (dispatch, getState) => {
```
Comment (minotogna) [inline webapp/loggedin/modules/analysis/processingChain/actions.js]: can this be called initProcessingChain?

File: webapp/loggedin/modules/analysis/processingChain/processingChainState.js [hunk 3]
```diff
@@ -21,9 +22,19 @@ const getProcessingChainOrig = _getStateProp(keys.orig, {})
...
 
-export const assocProcessingChain = chain => R.pipe(R.assoc(keys.orig, chain), R.assoc(keys.dirty, chain))
+export const assocProcessingChain = (chain, attributeUuidsOtherChains = null) =>
+  R.pipe(
+    R.assoc(keys.orig, chain),
+    R.assoc(keys.dirty, chain),
+    R.unless(
+      R.always(R.isNil(attributeUuidsOtherChains)),
+      R.assoc(keys.attributeUuidsOtherChains, attributeUuidsOtherChains),
+    ),
```
Comment (minotogna) [inline webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: do you really need unless? what if you assoc null?

File: server/modules/analysis/repository/processingStepCalculationRepository.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingStepCalculationRepository.js]: ok, sure

File: webapp/loggedin/modules/analysis/processingChain/actions.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingChain/actions.js]: ok

File: webapp/loggedin/modules/analysis/processingChain/processingChainState.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: assocProcessingChain is called even by saveDirty, maybe we can avoid that call and remove the "unless" check

Comment (minotogna) [reply webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: then you need two different assoc methods, one init and one assoc

File: server/modules/analysis/repository/processingStepCalculationRepository.js (hunk 1 above)
Comment (minotogna) [reply server/modules/analysis/repository/processingStepCalculationRepository.js]: maybe fetchCalculationAttributeUuidsByChainUuidExcluded ?

File: webapp/loggedin/modules/analysis/processingChain/processingChainState.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: good

File: webapp/loggedin/modules/analysis/processingChain/processingChainState.js [hunk 4]
```diff
@@ -21,9 +22,16 @@ const getProcessingChainOrig = _getStateProp(keys.orig, {})
...
 
+export const getAttributeUuidsOtherChains = _getStateProp(keys.attributeUuidsOtherChains, [])
+
 // ====== UPDATE
 
-export const assocProcessingChain = chain => R.pipe(R.assoc(keys.orig, chain), R.assoc(keys.dirty, chain))
+export const assocProcessingChain = (chain, attributeUuidsOtherChains) =>
+  R.pipe(
+    R.assoc(keys.orig, chain),
+    R.assoc(keys.dirty, chain),
```
Comment (minotogna) [inline webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: can you have one method _assocChain and be called by this method and saveDirty?

Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingChain/processingChainState.js]: sure

File: server/modules/analysis/repository/processingStepCalculationRepository.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingStepCalculationRepository.js]: ok

**Pull request merged**